packages/venia-concept/src/components/ProductImageCarousel/carousel.js

```diff
@@ -4,34 +4,62 @@ import PropTypes from 'prop-types';
 import classify from 'src/classify';
 import ThumbnailList from './thumbnailList';
 import defaultClasses from './carousel.css';
-
-const uri =
-    'iVBORw0KGgoAAAANSUhEUgAAAAQAAAAFCAQAAADIpIVQAAAADklEQVR42mNkgAJGIhgAALQABsHyMOcAAAAASUVORK5CYII=';
+import { makeProductMediaPath } from 'src/util/makeMediaPath';
+import { transparentPlaceholder } from 'src/shared/images';
 
 class Carousel extends Component {
     static propTypes = {
         classes: PropTypes.shape({
-            currentImage: PropTypes.string,
+            currentImageFile: PropTypes.string,
             root: PropTypes.string
         }),
         images: PropTypes.arrayOf(
             PropTypes.shape({
-                id: PropTypes.string
+                label: PropTypes.string,
+                position: PropTypes.number.isRequired,
+                disabled: PropTypes.bool,
+                file: PropTypes.string.isRequired
             })
-        )
+        ).isRequired
     };
 
+    shouldComponentUpdate({ images }) {
+        return (
+            images.length !== this.props.images.length ||
+            this.props.images.some(({ label, position, disabled, file }, i) => {
+                const image = images[i];
+                return (
+                    image.label !== label ||
+                    image.position !== position ||
+                    image.disabled !== disabled ||
+                    image.file !== file
+                );
+            })
+        );
+    }
+
     render() {
         const { classes, images } = this.props;
+        // the order of the array is not guaranteed to be the position order,
```

Comment:
Can we link to the issue you mentioned creating to ensure the API returns a sorted list?